src/components/Router/index.js

```diff
@@ -98,12 +99,18 @@ class CrossDeviceMobileRouter extends Component {
       return this.setError()
     }
     this.setState({token, steps, step, loading: false})
+    if (language) this.resetI18n(language)
```

Comment:
This is good.

Possibly out of scope, but have you considered:

- The language of the SMS
- The default country flag in the SMS input